### PR TITLE
dpdk - fix checking if any packet has actually been parsed

### PR DESCRIPTION
**Pull request by TheSableCZ**

### CODE REVIEW

File: input/dpdk.cpp [hunk 1]
```diff
@@ -292,7 +292,7 @@ InputPlugin::Result DpdkReader::get(PacketBlock& packets)
     m_stats.receivedPackets += receivedPackets;
     m_stats.receivedBytes += packets.bytes;
 
-    return Result::PARSED;
+    return opt.pblock->cnt ? Result::PARSED : Result::NOT_PARSED;
```
Comment (hynekkar) [inline input/dpdk.cpp:295]: The opt structure is not defined when WITH_FLEXPROBE is defined. This should be fixed.

Comment (TheSableCZ) [reply input/dpdk.cpp]: Fixed

Comment (hynekkar) [reply input/dpdk.cpp]: Perfect! Thanks!

**Comment by Lukas955** [comment]
> This solution is not correct.
> 
> Let's analyze it, dpdk-ring for simplicity:
> 
> 1. we create opt with PacketBlock, I expect the opt.pblock->cnt is analyzed to 0
> 2. loop L200 to L209 iterates over i, which is not used inside parse_packet
> 3. let's assume we received 10 packets -> `pkts_read_ = 10`
> 4. let's assume packet 0. is unknown not parsed
> 5. parse_packet() uses `opt->pblock->cnt` (probably initialized to 0) as index to the array `opt->pblock->pkts`
> 6. for packet 0 we can return, e.g., L725 or L737 (parser.cpp) but nothing changes `opt->pblock->cnt` (increment is L775)
> 7. for the next packets until pkts_read_ we are still on the same packet and nothing is processed - we loose everything -> this is not expected behavior and this fix cannot solve it.
>    It is worth noting that if any packet index > 0 and < pkts_read_ is unknown/invalid and parsers returns without incrementing `opt->pblock->cnt` we get Result::PARSED from `return opt.pblock->cnt ? Result::PARSED : Result::NOT_PARSED;` L214, but  any packet after the unknown one is skipped (because opt->pblock->cnt is not incremented).
> 
> Is it ensured somewhere, that `opt->packet_valid` is initialized to false? Is it used somewhere to distinguish theparsed and not_parsed individual packets? (the packetblock may contain both valid and invalid)

I think the solution in this pull request IS correct.

ad 1) pblock->cnt is initialized to zero [here](https://github.com/CESNET/ipfixprobe/blob/ebb8d77d7fee8ef8046a219aaebad386630612ff/workers.cpp#L61)  workers.cpp:61
ad 6) Not incrementing counter is correct behaviour. This ensures that packets that cannot be parsed by the parser, are ignored. 
ad 7) You interpretation is not correct. The next packet in the dpdk-ring [loop](https://github.com/CESNET/ipfixprobe/blob/ebb8d77d7fee8ef8046a219aaebad386630612ff/input/dpdk-ring.cpp#L200) is passed to the parser and if it is successfully parsed, the counter will be incremented [here](https://github.com/CESNET/ipfixprobe/blob/ebb8d77d7fee8ef8046a219aaebad386630612ff/input/parser.cpp#L775). Keep on mind that PacketBlock filled by the parser is different from burst received by the input module. In other words, only successfully parsered packets are placed in the PacketBlock. 

`opt->packet_valid` is something old and used only by PCAP input module.

**Review by hynekkar — APPROVED** [review]

**Comment by cejkato2** [comment]
Perfect teamwork! :-)

Ok, I found the most confusing that other parse functions had doxygen comment and their input parameters are the first ones. Contrary, `parse_packet()` unfortunately has the first parameter `opt` as output and the doxygen was missing at all...
I created some draft of the doc, maybe it will help with the future investigation. Anyway, this part of the source codes is a candidate for refactoring...

Additionally, I've created an update for the new package release and I'm waiting for a test build on COPR in @CESNET/NEMEA-testing.

**Review by cejkato2 — APPROVED** [review]

**Pull request merged**